blog/content/post/2021-10-28-secure-jtag-on-imx-soc.md

```diff
@@ -0,0 +1,270 @@
+---
+title: 'Securing JTAG on i.MX6 and i.MX8MM'
+abstract: 'JTAG help a lot of engineers during product development.
+          It also may be helpful for hackers.
+          I tell you why and how to increase JTAG security in your product'
+cover: /covers/secure-jtag.png
+author: michal.kotyla
+layout: post
+published: true
+date: 2021-10-28
+archives: "2021"
+
+tags:
+  - security
+  - jtag
+  - fuse
+  - bit
+  - fusebit
+  - fusebits
+  - fusing
+  - fuses
+  - nxp
+  - imx
+  - imx6
+  - i.mx6
+  - imx8
+  - i.mx8
+  - imx8mm
+  - i.mx8mm
+categories:
+  - Firmware
+  - Miscellaneous
+  - OS Dev
+  - Security
+  - Manufacturing
+
+---
+## Introduction
+
+JTAG port is an important feature that help a lot of engineers during product
+development. It also may be helpful for hackers while reverse-engineering
+process - e.g. they can dump the memory of your firmware and get access to
+strictly confidential information. In this article, I show you how to lock JTAG
+access for users who don't have a special key (password). I use for that Secure
+JTAG feature which is implemented on i.MX SoC's. In our lab, we tested this
+solution on i.MX6 and i.MX8MM. Probably this same feature is available on i.MX8
```

Comment:
Lets point here that we were able to achieve this on imx6, on the rest there is some missing documentation

Reply:
ok